planetaryimage/pds3image.py

```diff
@@ -136,7 +136,7 @@ def _save(self, file_to_write, overwrite):
         image_pointer = int(label_sz / self.label['RECORD_BYTES']) + 1
         self.label['^IMAGE'] = image_pointer + 1
 
-        if self._sample_bytes != self.data.itemsize:
+        if self._sample_bytes != self.label['IMAGE']['SAMPLE_BITS'] * 8:
```

Comment:
Now, as `self._sample_bytes` and `self.data.itemsize` are equal, we are comparing it with the `label` information. So, if our `sample_bytes` do not match with the `label` information, we will have to change the label information.